src/puppetlabs/trapperkeeper/bootstrap.clj

```diff
@@ -8,16 +8,20 @@
             [puppetlabs.trapperkeeper.services :as services]
             [puppetlabs.trapperkeeper.common :as common]
             [schema.core :as schema]
-            [me.raynes.fs :as fs]))
+            [me.raynes.fs :as fs]
+            [slingshot.slingshot :refer [try+ throw+]]))
 
 ;; Schemas
-(def BootstrapLine {:namespace schema/Str :service-name schema/Str})
+(def ParsedBootstrapEntry {:namespace schema/Str :service-name schema/Str})
 (def BootstrapFiles [(schema/protocol io/IOFactory)])
+(def AnnotatedBootstrapEntry {:entry schema/Str
+                    :bootstrap-file schema/Str
+                    :line-number schema/Int})
 
 ;; Constants
 (def bootstrap-config-file-name "bootstrap.cfg")
 
-(schema/defn ^:private parse-bootstrap-line! :- BootstrapLine
```

Comment:
Ah, sorry I didn't mean to leave this change in. Agreed on adding the comment blocks. It sounds like in general we don't like marking things `^:private` since you can't require them in test files. I'll reorganize the file. Should I just get rid of all the private metadata tags?